ndsl/dsl/gt4py_utils.py

```diff
@@ -257,6 +262,25 @@ def _make_storage_data_3d(
     return buffer
 
 
+def _make_storage_data_4d(
```

Comment:
Can we generalize this with something like
```python
def _make_storage_data_Nd(
    data: Field,
    shape: Tuple[int, ...],
    start: Tuple[int, ...],
    *,
    backend: str,
) -> Field:
    buffer = zeros(shape, backend=backend)
    idx = tuple([slice(start, start + shape_dim) for shape_dim in data.shape])
    buffer[idx] = asarray(data, type(buffer))
    return buffer
```

and we just have to pass a default start if it doesn't exist (that can be build with something like `tuple([0]*data.ndim)`